packages/contracts/src/contracts/current/protocol/AssetProxyOwner/AssetProxyOwner.sol

```diff
@@ -0,0 +1,119 @@
+
+    bytes4 constant REMOVE_AUTHORIZED_ADDRESS_SELECTOR = bytes4(keccak256("removeAuthorizedAddress(address)"));
+
+    modifier validRemoveAuthorizedAddressTx(uint256 transactionId) {
+        Transaction storage tx = transactions[transactionId];
+        require(isAssetProxyRegistered[tx.destination]);
+        require(isFunctionRemoveAuthorizedAddress(tx.data));
+        _;
+    }
+
+    /// @dev Contract constructor sets initial owners, required number of confirmations,
+    ///      time lock, and list of AssetProxy addresses.
+    /// @param _owners List of initial owners.
+    /// @param _required Number of required confirmations.
+    /// @param _secondsTimeLocked Duration needed after a transaction is confirmed and before it becomes executable, in seconds.
+    /// @param _assetProxyContracts Array of AssetProxy contract addresses.
+    function AssetProxyOwner(
+        address[] memory _owners,
+        uint256 _required,
+        uint256 _secondsTimeLocked,
+        address[] memory _assetProxyContracts)
+        public
+        MultiSigWalletWithTimeLock(_owners, _required, _secondsTimeLocked)
+    {
+        for (uint256 i = 0; i < _assetProxyContracts.length; i++) {
+            require(_assetProxyContracts[i] != address(0));
+            isAssetProxyRegistered[_assetProxyContracts[i]] = true;
+        }
+    }
+
+    /// @dev Registers or deregisters an AssetProxy to be able to execute
+    ///      removeAuthorizedAddress without a timelock.
+    /// @param assetProxyContract Address of AssetProxy contract.
+    /// @param isRegistered Status of approval for AssetProxy contract.
+    function registerAssetProxy(address assetProxyContract, bool isRegistered)
+        public
+        onlyWallet
+        notNull(assetProxyContract)
+    {
+        isAssetProxyRegistered[assetProxyContract] = isRegistered;
+        AssetProxyRegistration(assetProxyContract, isRegistered);
+    }
+
+    /// @dev Allows execution of removeAuthorizedAddress without time lock.
+    /// @param transactionId Transaction ID.
+    function executeRemoveAuthorizedAddress(uint256 transactionId)
+        public
+        notExecuted(transactionId)
+        fullyConfirmed(transactionId)
+        validRemoveAuthorizedAddressTx(transactionId)
+    {
+        Transaction storage tx = transactions[transactionId];
+        tx.executed = true;
+        if (tx.destination.call.value(tx.value)(tx.data))
+            Execution(transactionId);
+        else {
+            ExecutionFailure(transactionId);
+            tx.executed = false;
+        }
+    }
+
+    /// @dev Compares first 4 bytes of byte array to removeAuthorizedAddress function signature.
+    /// @param data Transaction data.
+    /// @return Successful if data is a call to removeAuthorizedAddress.
+    function isFunctionRemoveAuthorizedAddress(bytes memory data)
+        public
+        pure
+        returns (bool)
+    {
+        bytes4 first4Bytes = readFirst4(data);
```

Comment:
Nit - Variable name.
More descriptive to use something like `bytes4 functionSelector = readFirst4(data);`.

Reply:
Going to move this to `LibBytes` and keep the variable name generic.